sample_functions/README.md

```diff
@@ -1,15 +1,56 @@
-Sample functions
-================
+# Sample functions
 
 Sampler which counts the number of times we find ourselves in a particular function. 
 Outputs a list of functions with their counts.
-	
-	Usage: 
-	  usample <start_bbcount> <end_bbcount> <bbcount_interval> 
-	  Parameters:
-		start_bbcount: Time, given as a bbcount to start sampling.
-		end_bbcount: Last time that may be sampled.
-		bbcount_interval: How often to take a sample, in number of basic blocks.
-
-	  E.g. 1 1000 1 
-	  means sample every basic block from 1 to 1000. 
+```
+Usage:
+  usample <start_bbcount> <end_bbcount> <bbcount_interval> [<filename>]
+  Parameters:
+    start_bbcount: Time, given as a basic block count to start sampling.
+    end_bbcount: Last time that may be sampled.
+    bbcount_interval: How often to take a sample, in basic block count.
+    filename: A file to output the sampled stacks to.
+
+  E.g. 1 1000 1
+  means sample every basic block count from 1 to 1000.
+```
+
+## Generating flame graphs
+
+The output from this tool can be used directly to generate [flame graphs](https://www.brendangregg.com/flamegraphs.html).
+These graphs provide a visual indication of the busiest areas of your program.
+Generally, around 1,000 samples will produce a useful flame graph. You may
+wish to limit the sampling to a specific range, for example after start up has
+completed.
+
+To generate a flame graph:
+ - clone the [flame graph repository](https://github.com/brendangregg/FlameGraph)
+
+ - run the `usample` script. In this case we first query the recorded time
+ range within UDB:
+```
+(udb) info time
```

Comment:
Can you please use `> ` as prompt as we don't use `(udb) ` any more (but the full status prompt is not needed).